src/profile-logic/tracks.js

```diff
@@ -394,36 +394,56 @@ export function getLocalTrackName(
   }
 }
 
+// Any thread with less than 1% non-idle time will be hidden.
+const PERCENTAGE_ACTIVE_SAMPLES = 0.01;
+
 /**
  * Determine if a thread is idle, so that it can be hidden. It is really annoying for an
- * end user to load a profile full of empty and idle threads.
+ * end user to load a profile full of empty and idle threads. This function goes through
+ * all of the samples in the thread, and sees if some large percentage of them are idle.
  */
-function _isThreadIdle(thread: Thread): boolean {
-  // Hide content threads with no RefreshDriverTick. This indicates they were
-  // not painted to, and most likely idle. This is just a heuristic to help users.
-  if (thread.name === 'GeckoMain' && thread.processType === 'tab') {
-    let isPaintMarkerFound = false;
-    if (thread.stringTable.hasString('RefreshDriverTick')) {
-      const paintStringIndex = thread.stringTable.indexForString(
-        'RefreshDriverTick'
-      );
+function _isThreadIdle(profile: Profile, thread: Thread): boolean {
+  if (
+    // Don't hide the compositor.
+    thread.name === 'Compositor' ||
+    // Don't hide the main thread.
+    (thread.name === 'GeckoMain' && thread.processType === 'default')
+  ) {
+    return false;
+  }
 
-      for (
-        let markerIndex = 0;
-        markerIndex < thread.markers.length;
-        markerIndex++
-      ) {
-        if (paintStringIndex === thread.markers.name[markerIndex]) {
-          isPaintMarkerFound = true;
-          break;
+  let maxActiveStackCount = PERCENTAGE_ACTIVE_SAMPLES * thread.samples.length;
+  let activeStackCount = 0;
+  let filteredStackCount = 0;
+
+  for (
+    let sampleIndex = 0;
+    sampleIndex < thread.samples.length;
+    sampleIndex++
+  ) {
+    const stackIndex = thread.samples.stack[sampleIndex];
+    if (stackIndex === null) {
+      // This stack was filtered out. Most likely this will never actually happen
+      // on a new profile, but keep this check here since the stacks are possibly
+      // null in the Flow type definitions.
+      filteredStackCount++;
+      // Adjust the maximum necessary active stacks to find based on null stacks.
+      maxActiveStackCount =
+        PERCENTAGE_ACTIVE_SAMPLES *
+        (thread.samples.length - filteredStackCount);
+    } else {
+      const categoryIndex = thread.stackTable.category[stackIndex];
+      const category = profile.meta.categories[categoryIndex];
+      if (category.name !== 'Idle') {
+        activeStackCount++;
+        if (activeStackCount > maxActiveStackCount) {
+          return false;
         }
       }
     }
-    if (!isPaintMarkerFound) {
-      return true;
-    }
   }
-  return false;
+
+  return true;
```

Comment:
Nit: This is super minor since the comment above points out that the first branch will likely never happen, and the scenario I'm pointing out here is even less likely to happen.

But, if `maxActiveStackCount` is reduced in the loop, it will only be acted on in the second branch, and so if all remaining iterations in the loop enters the first branch (that is, all remaining samples are filtered), it is possible that the threshold could be breached as the limit shrinks, but it never hits the `return false` since the loop is exited before that. Replacing this line with `return activeStackCount <= maxActiveStackCount` would account for that scenario.